bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/i18n/ThingStatusInfoI18nLocalizationService.java

```diff
@@ -98,35 +98,76 @@ protected void unsetTranslationProvider(TranslationProvider i18nProvider) {
         this.i18nProvider = null;
     }
 
+    /**
+     * Returns the translation of the description for the specified locale, using the translations from the bundles of
+     * the given thingHandler and its parent classes. The description may contain arguments that may also need
+     * translation (see class JavaDoc for an example); those arguments are translated in the same way.
+     */
+    private String translateDescription(@NonNull String description, @Nullable Locale locale,
+            @NonNull ThingHandler thingHandler) {
+        ParsedDescription parsedDescription = new ParsedDescription(description);
+
+        Object[] translatedArgs = null;
+        if (parsedDescription.args != null) {
+            translatedArgs = Arrays.stream(parsedDescription.args).map(arg -> {
+                if (I18nUtil.isConstant(arg)) {
+                    return getTranslationForClass(arg, locale, thingHandler.getClass());
+                } else {
+                    return arg;
+                }
+            }).toArray(String[]::new);
+        }
+
+        return getTranslationForClass(parsedDescription.key, locale, thingHandler.getClass(), translatedArgs);
+    }
+
+    /**
+     * Returns the translation for the given i18n constant and locale using the translations from the bundle of the
+     * given class; if there is no translation look up the translation in the bundle in the parent class, and so
+     * forth. If no translation is found for the bundle of any parent class, return the i18n constant.
+     */
+    @NonNull
+    private String getTranslationForClass(@NonNull String i18nConstant, @Nullable Locale locale,
+            @Nullable Class<?> clazz, @Nullable Object... args) {
+        if (clazz == null) {
+            return i18nConstant;
+        }
+
+        Bundle bundle = FrameworkUtil.getBundle(clazz);
+
+        if (bundle == null) {
+            return getTranslationForClass(i18nConstant, locale, clazz.getSuperclass());
```

Comment:
Why don't you pass the `args` as a parameter to this method call?

Reply:
Good question :) The answer is that I simply forgot it (probably because this is a corner case for which I cannot imagine how it could happen - this case is relevant if, in the class hierarchy of the thing handler, there is a superclass not defined by a bundle class loader, which has a superclass within a bundle with i18n translations).

Thanks for pointing this out.